### DISCUSSION TITLE
Run other action after fixes were committed

### DISCUSSION THREAD

**Discussion by FlorianPfaff** [oxsecurity/megalinter#2958] [Q&A] [answered]
Hi,

maybe one of the avid users here knows what to do: I have required tests in my branch protection rules. If MegaLinter does not apply any fixes, everything works as expected. If MegaLinter applies fixes, the tests are not run.

![image](https://github.com/oxsecurity/megalinter/assets/6773539/25c5885e-0794-4dad-9f4d-95eca40cc828)
I was thinking about different possibilities to make the tests run. I added a on
```
  workflow_run:
    workflows: ["MegaLinter"] # Name of the MegaLinter workflow
    types:
      - completed

```
trigger to run [the test after megalinter finished](https://github.com/FlorianPfaff/pyRecEst/blob/megalinteractiontest-new/.github/workflows/tests.yml) but it does not work. I also tried giving megalinter the [actions permission](https://github.com/FlorianPfaff/pyRecEst/blob/megalinteractiontest-new/.github/workflows/mega-linter.yml) because I thought maybe then MegaLinter will also be able to trigger the test action via its commit. Still, no luck.

Is here anyone else who experienced this issue? I would appreciate hints on how to solve this problem. Thanks!

**Comment by Kurt-von-Laven** [comment]
GitHub doesn't allow bots to trigger workflows for security reasons and to prevent infinite loops. We [run MegaLinter as a pre-commit hook](https://megalinter.io/latest/mega-linter-runner/#pre-commit-hook) instead and don't push automatic fixes when running MegaLinter in CI. You could alternatively pull down the changes MegaLinter made, squash them into the appropriate commits, and force-push the branch. You could also trigger the test workflow with a PAT, although that will make your CI pipeline more vulnerable to attackers.

**Comment by nvuillam** [comment, marked as answer]
@FlorianPfaff the only way to trigger again jobs is to define a Personal Access Token and store it in PAT value 

You can see how to configure it in MegaLinter documentation :) -> https://megalinter.io/latest/config-apply-fixes/#apply-fixes-issues
